src/mgmt/srv.c

```diff
@@ -409,6 +409,73 @@ process_join_request(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 	D_FREE(resp);
 }
 
+/*
+ * Adds user and group name as owner to a new copy of the daos_prop_t passed in.
+ * The newly allocated prop is freed after pool create call returns.
+ */
+static int
+add_ownership_props(daos_prop_t **prop_out, char *owner, char *owner_grp)
+{
+	daos_prop_t    *final_prop = NULL;
+	char	       *out_owner = NULL;
+	char	       *out_owner_grp = NULL;
+	uint32_t	idx = 0;
+	uint32_t	entries = 0;
+	int		rc = 0;
+
+	if (owner != NULL && *owner != '\0') {
+		D_ASPRINTF(out_owner, "%s", owner);
+		if (out_owner == NULL) {
+			D_ERROR("Failed to allocate string for owner\n");
+			rc = -DER_NOMEM;
+			goto err_out;
+		}
+
+		entries++;
+	}
+
+	if (owner_grp != NULL && *owner_grp != '\0') {
+		D_ASPRINTF(out_owner_grp, "%s", owner_grp);
+		if (out_owner_grp == NULL) {
+			D_ERROR("Failed to allocate string for owner_grp\n");
+			rc = -DER_NOMEM;
+			goto err_out;
+		}
+
+		entries++;
+	}
+
```

Comment:
If neither owner-user nor owner-group was provided, I think we can quit right here and avoid allocating the prop.